core/trino-main/src/main/java/io/trino/type/Json2016Type.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.type;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.airlift.slice.Slice;
+import io.trino.spi.block.Block;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.type.AbstractVariableWidthType;
+import io.trino.spi.type.StandardTypes;
+import io.trino.spi.type.TypeSignature;
+
+import static io.airlift.slice.Slices.utf8Slice;
+
+public class Json2016Type
+        extends AbstractVariableWidthType
+{
+    public static final Json2016Type JSON_2016 = new Json2016Type();
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    public Json2016Type()
+    {
+        super(new TypeSignature(StandardTypes.JSON_2016), JsonNode.class);
+    }
+
+    @Override
+    public Object getObjectValue(ConnectorSession session, Block block, int position)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void appendTo(Block block, int position, BlockBuilder blockBuilder)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Object getObject(Block block, int position)
+    {
+        if (block.isNull(position)) {
+            return null;
+        }
+
+        Slice bytes = block.getSlice(position, 0, block.getSliceLength(position));
+        try {
+            return MAPPER.readTree(bytes.toStringUtf8());
+        }
+        catch (JsonProcessingException e) {
+            // TODO
+            throw new RuntimeException(e);
```

Comment:
looks like you can already fill this in in `Analyze ... `commit

Reply:
applied to both `getObject` and `writeObject` methods